lib/chef/knife/core/bootstrap_context.rb

```diff
@@ -213,7 +213,7 @@ def first_boot
               attributes[:run_list] = @run_list
             end
 
-            attributes.delete(:run_list) if attributes[:policy_name] && !attributes[:policy_name].empty?
+            attributes.delete(:run_list) if ( attributes[:policy_name] && !attributes[:policy_name].empty? ) || ( attributes["policy_name"] && !attributes["policy_name"].empty? )
```

Comment:
Can you add a comment here so that we know why we're checking both the symbol and the string varients

Reply:
When we pass policy_name via bootstrap options i.e `--policy-name` then we need to check it by symbol.
And when we pass policy_name by `--json-attribute-file` it comes in string. Hence both check is added.

Reply:
They should just be strings, since we're coming from parsing JSON the symbols shouldn't ever be set.

Alternatively, the config setting should be a Mash so it doesn't matter.

Reply:
Thanks @tas50 and @lamont-granquist,  When we pass `policy_name` via bootstrap options i.e `--policy-name` then we need to check it by symbol and when we pass `policy_name` by `--json-attribute-file` it comes in string. Hence both check is added.
I was trying to change config into Mash but then it will affect other bootstrap options which may impact a lot. looking into more if there is any another possible way to do so.